### PR TITLE
automate building docker images

### PR DESCRIPTION
**Pull request by mistafunk**
* Deduplicated the Dockerfiles (parametrized Python version)
* Introduced pyprt-base images to extract the heavy installation of the VS build tools (C++ compiler)

### CODE REVIEW

**Comment by clechot** [comment]
Suggestion: would it be possible to deduplicate the yml files? I noticed that for one Python version, the yml environment files from each platform were the same. This has been done in the PyPRT examples repo.

**Comment by mistafunk** [comment]
> Suggestion: would it be possible to deduplicate the yml files? I noticed that for one Python version, the yml environment files from each platform were the same. This has been done in the PyPRT examples repo.

Did not find an easy way to parametrize conda env files... let's do that in a separate issue.

**Review by clechot — CHANGES_REQUESTED** [review]
Could you also change the README at line 87 `devtoolset-9` instead of 8?

File: README.md [hunk 1]
```diff
@@ -91,11 +91,11 @@ _Note: on Windows, replace `bin` with `Scripts` in the following commands. Some
 ### Build Python Wheel
 
 1. Open a shell in the PyPRT git root.
-1. First time only: set up a virtual Python environment with build dependencies for PyPRT. Adapt `python3.6` and `centos7/py36` to your desired OS/Python combination.
+1. First time only: set up a virtual Python environment with build dependencies for PyPRT. Adapt `python3.6` and the `envs` file to your desired OS/Python combination.
     1. Create the virtual environment: `python3.6 -m venv .venv`
     1. Get latest pip: `.venv/bin/python -m pip install --upgrade pip`
     1. Get latest wheel: `.venv/bin/python -m pip install --upgrade wheel`
-    1. Install build dependencies for PyPRT: `.venv/bin/python -m pip install -r envs/centos7/py36/requirements.txt`
+    1. Install build dependencies for PyPRT: `.venv/bin/python -m pip install -r envs/centos7/wheel/requirements-py36.txt`
```
[the remark this answers is not in the document]
Comment (mistafunk) [reply README.md:98]: done

File: README.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ _Note: on Windows, replace `bin` with `Scripts` in the following commands. Some
 1. First time only: run `conda env create --prefix ./env --file envs/centos7/py36-conda/environment.yml` to create a conda environment with all the required Python packages (adapt `centos7/py36-conda` to your desired OS/Python combination).
```
[the remark this answers is not in the document]
Comment (mistafunk) [reply README.md:106]: done

File: README.md [hunk 3]
```diff
@@ -106,7 +106,7 @@ _Note: on Windows, replace `bin` with `Scripts` in the following commands. Some
...
 1. Run `activate ./env`.
 1. Run `python setup.py bdist_conda`. This will build the CMake project and Python packages.
-1. The resulting package is written to the `./env/conda-bld/{platform}` folder.
+1. The resulting package is written to the `./env-py36/conda-bld/{platform}` folder.
```
[the remark this answers is not in the document]
Comment (mistafunk) [reply README.md:109]: good catch, done

File: README.md [hunk 4]
```diff
@@ -158,9 +158,9 @@ Note: We only support Docker on Linux and Windows. On Windows, Docker needs to b
 #### Build Wheels
 
 1. Open a shell in the PyPRT git root
-1. Create the desired image for the build toolchain (adapt `py36` to your desired Python version):
-   * Linux: `docker build --rm -f envs/centos7/py36/Dockerfile -t pyprt:centos7-py36 --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g) .`
-   * Windows: `docker build --rm -f envs\windows\py36\Dockerfile -t pyprt:windows-py36 .`
+1. Create the desired image for the build toolchain (adapt to your desired Python version):
+   * Linux: `docker build --rm -f envs/centos7/wheel/Dockerfile -t pyprt:centos7-py36 --build-arg PY_VER=36 --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g) .`
+   * Windows: `docker build --rm -f envs\windows\wheel\Dockerfile -t pyprt:windows-py36 --build-arg PY_VER=3.6 .`
```
[the remark this answers is not in the document]
Comment (mistafunk) [reply README.md:163]: good catch, done

File: README.md [hunk 5]
```diff
@@ -170,8 +170,8 @@ Note: We only support Docker on Linux and Windows. On Windows, Docker needs to b
 
 1. Open a shell in the PyPRT git root
 1. Create the desired image for the build toolchain (adapt `py36` to your desired Python version):
-    * Linux: `docker build --rm -f envs/centos7/py36-conda/Dockerfile -t pyprt:centos7-py36-conda --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g) .`
-    * Windows: `docker build --rm -f envs\windows\py36-conda\Dockerfile -t pyprt:windows-py36-conda .`
+    * Linux: `docker build --rm -f envs/centos7/conda/Dockerfile -t pyprt:centos7-py36-conda --build-arg PY_VER=36 --build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g) .`
+    * Windows: `docker build --rm -f envs\windows\conda\Dockerfile -t pyprt:windows-py36-conda --build-arg PY_VER=3.6 .`
```
[the remark this answers is not in the document]
Comment (mistafunk) [reply README.md:174]: done

**Review by clechot — APPROVED** [review]

**Pull request merged**